### PR TITLE
Create iready.tests_by_round

### PR DESCRIPTION
**Pull request by anthonygwalters**
# Pull Request

## If approved, this pull request will

- <feature/fix 1>
- <feature/fix 2>
- ...

## Protocol

- [ ] Go here: [gabby - Asana](https://app.asana.com/0/35737884613515/list)
- [ ] Click <kbd>+ Add new</kbd> > <kbd>Pull Request</kbd>
- [ ] Re-name the task to your pull request title
- [ ] Assign a due date, typically Thursday
- [ ] If this is a hotfix, please mark it as such
- [ ] Review the code checks listed in the task
- [ ] Paste the link from the task you created in the space below:

## Asana Link
https://app.asana.com/0/35737884613515/1204771377996029/f

### CODE REVIEW

File: gabby/iready/view/tests_by_round.sql [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      ) AS rn_subj_year
+    FROM
+      gabby.iready.diagnostic_results AS dr
+      LEFT JOIN gabby.people.school_crosswalk AS sc ON (dr.school = sc.site_name)
+      LEFT JOIN gabby.reporting.reporting_terms AS rt ON (
+        (
+          dr.completion_date BETWEEN rt.start_date AND rt.end_date
+        )
+        AND rt.identifier = 'IR'
+        AND sc.region = rt.region
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:80]: is this 1 RT record per region?

File: gabby/iready/view/tests_by_round.sql [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  ir.rn_subj_round,
+  ir.rn_subj_year
+FROM
+  iready AS ir
+  LEFT JOIN gabby.assessments.fsa_iready_crosswalk AS cwo ON (
+    ir.overall_scale_score BETWEEN cwo.scale_low AND cwo.scale_high
+    AND ir.[subject] = cwo.test_name
+    AND ir.grade_level = cwo.grade_level
+    AND cwo.source_system = 'i-Ready'
+    AND CASE
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:129]: refactor to iready CTE

File: gabby/iready/view/tests_by_round.sql [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        WHEN dr.overall_relative_placement = '2 Grade Levels Below' THEN 2
+        WHEN dr.overall_relative_placement = '1 Grade Level Below' THEN 3
+        WHEN dr.overall_relative_placement = 'Early On Grade Level' THEN 4
+        WHEN dr.overall_relative_placement = 'Mid or Above Grade Level' THEN 5
+      END AS orp_numerical,
+      CASE
+        WHEN dr.overall_relative_placement IN (
+          'Early On Grade Level',
+          'Mid or Above Grade Level'
+        ) THEN 'On or Above Grade Level' -- trunk-ignore(sqlfluff/LT05)
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:40]: rm comment

File: gabby/iready/view/tests_by_round.sql [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    AND ir.grade_level = cwo.grade_level
+    AND cwo.source_system = 'i-Ready'
+    AND CASE
+      WHEN ir.region IN ('KCNA', 'TEAM') THEN 'NJSLA'
+      WHEN ir.region = 'KMS' THEN 'FL'
+    END = cwo.destination_system
+  )
+  LEFT JOIN gabby.assessments.fsa_iready_crosswalk AS cwt ON (
+    -- trunk-ignore(sqlfluff/LT05)
+    ir.overall_scale_score + ir.annual_typical_growth_measure BETWEEN cwt.scale_low AND cwt.scale_high
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:136]: refactor addition to iready CTE

File: gabby/iready/view/tests_by_round.sql [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    AND ir.[subject] = cwt.test_name
+    AND ir.grade_level = cwt.grade_level
+    AND cwt.source_system = 'i-Ready'
+    AND CASE
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:140]: refactor to iready CTE

File: gabby/iready/view/tests_by_round.sql [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+      WHEN ir.region IN ('KCNA', 'TEAM') THEN 'NJSLA'
+      WHEN ir.region = 'KMS' THEN 'FL'
+    END = cwt.destination_system
+  )
+  LEFT JOIN gabby.assessments.fsa_iready_crosswalk AS cws ON (
+    -- trunk-ignore(sqlfluff/LT05)
+    ir.overall_scale_score + ir.annual_stretch_growth_measure BETWEEN cws.scale_low AND cws.scale_high
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:147]: refactor to iready CTE

File: gabby/iready/view/tests_by_round.sql [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    AND ir.[subject] = cws.test_name
+    AND ir.grade_level = cws.grade_level
+    AND cws.source_system = 'i-Ready'
+    AND CASE
```
Comment (cbini) [inline gabby/iready/view/tests_by_round.sql:151]: refactor to iready CTE

**Review by cbini — CHANGES_REQUESTED** [review]

File: gabby/iready/view/tests_by_round.sql (hunk 1 above)
Comment (anthonygwalters) [reply gabby/iready/view/tests_by_round.sql:80]: 3 rounds, one record per region, per year

File: gabby/iready/view/tests_by_round.sql (hunk 2 above)
Comment (cbini) [reply gabby/iready/view/tests_by_round.sql:129]: just meant the case statement

**Comment by anthonygwalters** [comment]
Can't fix "line too long" issue

**Review by cbini — APPROVED** [review]

**Pull request merged**